src/satellite/process.ts

```diff
@@ -532,14 +773,18 @@ export class SatelliteProcess implements Satellite {
     ]
 
     if (isAck) {
+      /* Currenly we do not clean oplog table on acknowledgement due
```

Comment:
I left a comment to have an opportunity to discuss this further. 

With current version of GC we will keep oplog entries for longer time frames then we used to have, 
which will causes extra performance penalty when syncing transactions from Electric, and extra space
consumed on the device as we do not merge entries in the oplog, and they going to be GCed only on 
receiving same transaction from Electric and not just when Electric have confirmed that transaction has
been written. Also sending the whole transaction back to the source looks like an overkill in this situation,
 so it's also something for us to consider for optimization.

Reply:
>  Also sending the whole transaction back to the source looks like an overkill in this situation,
so it's also something for us to consider for optimization.
We could just send a gist of the transaction and not the transaction itself, but we need to verify the correctness of that. Right now, the transaction that we receive might not be exactly the one that we've sent (it will be the result of applying it on the server against concurrent outstanding operations). We would have to ensure that there are no corner cases in skipping the transaction... I am not very concerned with avoiding transmitting the transaction, because we have to send it to all other satellites anyways.

> With current version of GC we will keep oplog entries for longer time frames
We can eventually compact the log by merging entries that are safe to merge and other strategies. Compaction would be very efficient at least in the worst case for growing the oplog, which is when the client is disconnected.

I hope that the oplog doesn't grow so large when the user is online, otherwise it will not be a very 'real-time' system. In that sense I believe that the cost of the oplog should be much smaller compared to the cost of keeping application data on the device.